ui/app/src/components/TransferList/TransferList.tsx

```diff
@@ -0,0 +1,208 @@
+    );
+  };
+
+  const moveLeftToRight = () => {
+    const nextCheckedList = {};
+    const leftCheckedItems = getChecked(sourceItems);
+
+    leftCheckedItems.forEach((item) => {
+      nextCheckedList[item.id] = false;
+    });
+
+    setCheckedList({ ...checkedList, ...nextCheckedList });
+    setSourceItems(not(sourceItems, leftCheckedItems));
+    setTargetItems([...targetItems, ...leftCheckedItems]);
+    onTargetListItemsAdded(leftCheckedItems);
+  };
+
+  const moveRightToLeft = () => {
+    const nextCheckedList = {};
+    const rightCheckedItems = getChecked(targetItems);
+
+    rightCheckedItems.forEach((item) => {
+      nextCheckedList[item.id] = false;
+    });
+
+    setCheckedList({ ...checkedList, ...nextCheckedList });
+    setTargetItems(not(targetItems, rightCheckedItems));
+    setSourceItems([...sourceItems, ...rightCheckedItems]);
+    onTargetListItemsRemoved(rightCheckedItems);
+  };
+
+  const isAllChecked = (items: TransferListItem[]) => {
+    return items.length
+      ? !items.some(
+          (item) =>
+            !Object.keys(checkedList).find(function(checked) {
+              return checked === item.id && checkedList[checked];
+            })
+        )
+      : false;
+  };
+
+  return (
+    <Box display="flex">
+      <TransferListColumn
+        title={props.source.title}
+        items={sourceItems}
+        checkedList={checkedList}
+        onCheckAllClicked={onCheckAllClicked}
+        onItemClick={onItemClicked}
+        isAllChecked={isAllChecked}
+        inProgressIds={inProgressIds}
+      />
+      <section className={classes.buttonsWrapper}>
+        <IconButton onClick={moveLeftToRight}>
+          <NavigateNextIcon />
+        </IconButton>
+        <IconButton onClick={moveRightToLeft}>
+          <NavigateBeforeIcon />
+        </IconButton>
+      </section>
+      <TransferListColumn
+        title={props.target.title}
+        items={targetItems}
+        checkedList={checkedList}
+        onCheckAllClicked={onCheckAllClicked}
+        onItemClick={onItemClicked}
+        isAllChecked={isAllChecked}
+        inProgressIds={inProgressIds}
+      />
+    </Box>
+  );
+}
+
+interface TransferListColumnProps {
+  title: ReactNode;
+  items: TransferListItem[];
+  onItemClick(item: TransferListItem): void;
+  checkedList: LookupTable<boolean>;
+  inProgressIds: string[];
+  isAllChecked(items: TransferListItem[]): boolean;
+  onCheckAllClicked(items: TransferListItem[], checked: boolean): void;
+}
+
+function TransferListColumn(props: TransferListColumnProps) {
+  const { title, items, onItemClick, checkedList, isAllChecked, onCheckAllClicked, inProgressIds } = props;
+  const classes = useStyles();
+  return (
+    <Paper className={classes.listPaper}>
+      <header className={classes.listHeader}>
+        <Checkbox checked={isAllChecked(items)} onChange={(event) => onCheckAllClicked(items, event.target.checked)} />
+        {title && <Typography color="textSecondary">{title}</Typography>}
```

Comment:
Make sure to test with a long title. How does title+searcbar behave in that case?